dali/python/backend_impl.cc

```diff
@@ -1648,6 +1650,22 @@ PYBIND11_MODULE(backend_impl, m) {
     .value("INTERP_GAUSSIAN", DALI_INTERP_GAUSSIAN)
     .export_values();
 
+  // DALIBorderType
+  py::enum_<boundary::BoundaryType>(types_m, "DALIBorderType", "Border policy.\n<SPHINX_IGNORE>")
+    .value("BORDER_CONSTANT", boundary::BoundaryType::CONSTANT,
+           "iiiiii|abcdefgh|iiiiiii with some specified i")
+    .value("BORDER_CLAMP", boundary::BoundaryType::CLAMP,
+           "aaaaaa|abcdefgh|hhhhhhh")
+    .value("BORDER_REFLECT_1001", boundary::BoundaryType::REFLECT_1001,
+           "fedcba|abcdefgh|hgfedcb")
+    .value("BORDER_REFLECT_101", boundary::BoundaryType::REFLECT_101,
+           "gfedcb|abcdefgh|gfedcba")
+    .value("BORDER_WRAP", boundary::BoundaryType::WRAP,
+           "cdefgh|abcdefgh|abcdefg")
+    .value("BORDER_TRANSPARENT", boundary::BoundaryType::TRANSPARENT,
+           "uvwxyz|abcdefgh|ijklmno")
+    .export_values();
+
```

Comment:
Recently, @klecki suggested to me that exposing enums in python is actually not a great idea. For once, it is very explicit protobuffer-wise, but most importantly @mzient and @jantonguirao seemed to agree that it is more user friendly to just ask the user for a string, like border="reflect_101". With explicit enum type you need to import `types`, know the enum type name, type the enum type name and then follow it with the name of the value. For string you just go with the value, and, even more, you can be so nice and parse for example some variants like "101", "reflcet101", "reflect_101", "REFLECT_101". In other words, enums are more diffucult for us to expose and for users to use.

Reply:
If you'd like you can just copy: https://github.com/NVIDIA/DALI/pull/4307/files#diff-03bec6668c5468dc08554a4de4020bfa60dfdaf4063966200fde14782801aeb0

Reply:
For the reference, for example take a look on the https://docs.nvidia.com/deeplearning/dali/user-guide/docs/operations/nvidia.dali.fn.resize.html#nvidia.dali.fn.resize and its `mode` param.

Reply:
After consideration, I'm going to remove the `DALIBorderType`.